code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -378,32 +392,32 @@ func (srh *StorageRestHandler) getAllocationBlobbers(w http.ResponseWriter, r *h
 	common.Respond(w, r, blobberIDs, nil)
 }
 
-func getBlobbersForRequest(request newAllocationRequest, edb *event.EventDb, balances cstate.TimedQueryStateContextI, limit common2.Pagination) ([]string, error) {
-	var sa = request.storageAllocation()
+func getBlobbersForRequest(request allocationBlobbersRequest, edb *event.EventDb, balances cstate.TimedQueryStateContextI, limit common2.Pagination) ([]string, error) {
+	var sa = StorageAllocation{}
 	var conf *Config
 	var err error
 	if conf, err = getConfig(balances); err != nil {
 		return nil, fmt.Errorf("can't get config: %v", err)
 	}
 
 	var creationDate = balances.Now()
-	sa.TimeUnit = conf.TimeUnit // keep the initial time unit
+	//sa.TimeUnit = conf.TimeUnit // keep the initial time unit
 
 	// number of blobbers required
-	var numberOfBlobbers = sa.DataShards + sa.ParityShards
+	var numberOfBlobbers = request.DataShards + request.ParityShards
 	if numberOfBlobbers > conf.MaxBlobbersPerAllocation {
 		return nil, common.NewErrorf("allocation_creation_failed",
 			"Too many blobbers selected, max available %d", conf.MaxBlobbersPerAllocation)
 	}
 
-	if sa.DataShards <= 0 || sa.ParityShards < 0 {
+	if request.DataShards <= 0 || request.ParityShards < 0 {
 		return nil, common.NewErrorf("allocation_creation_failed",
 			"invalid data shards:%v or parity shards:%v", sa.DataShards, sa.ParityShards)
 	}
 	// size of allocation for a blobber
-	var allocationSize = sa.bSize()
+	var allocationSize = int64(math.Ceil(float64(request.Size) / float64(request.DataShards)))
```

Comment:
we used sa here just not to copy-paste methods of sa here, not  we need to move common code to util package